src/handler/recalls.ts

```diff
@@ -0,0 +1,89 @@
+import { getProfile } from '@dvsa/cvs-feature-flags/profiles/vtx';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { MotSecret } from '../models/motRecalls';
+import { formatErrorMessage } from '../util/errorMessage';
+import { addHttpHeaders } from '../util/httpHeaders';
+import logger from '../util/logger';
+import { filterMotRecalls, getBearerToken, getMotRecallsByVin } from '../util/recalls';
+import { validateSingleVin } from '../validators/recalls';
+
+const cache: Map<string, string> = new Map();
+const defaultResponse = addHttpHeaders({
+  statusCode: 200,
+  body: JSON.stringify({
+    manufacturer: null,
+    hasRecall: false,
+  }),
+});
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Recalls end point called');
+  try {
+    const featureFlags = await getProfile();
+
+    if (!featureFlags.recallsApi) {
+      logger.error('Recall Feature Flag is undefined');
+      return addHttpHeaders({
+        statusCode: 500,
+        body: 'Recall Feature Flag is undefined',
+      });
+    }
+
+    if (!featureFlags.recallsApi.enabled) {
+      logger.warn('Flag disabled: please enable for recalls functionality');
+      return defaultResponse;
+    }
+
+    const vin: string = decodeURIComponent(event.pathParameters?.vin as string);
+    if (!validateSingleVin(vin)) {
+      logger.error(formatErrorMessage('VIN provided in path parameter is not valid.'));
+      return defaultResponse;
+    }
+
+    const motSecret = {
```

Comment:
I like how the getBearerToken logic sits in the utils, I think the assignment of these motSecret values could probably be delegated to there too, to keep this handler clean.